k8s-operator/sessionrecording/ws/message_test.go

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+//go:build !plan9
+
+package ws
+
+import (
+	"reflect"
+	"testing"
+
+	"go.uber.org/zap"
+)
+
+func Test_msg_Parse(t *testing.T) {
+	zl, err := zap.NewDevelopment()
+	if err != nil {
+		t.Fatalf("error creating a test logger: %v", err)
+	}
+	testMask := [4]byte{1, 2, 3, 4}
+	tests := []struct {
+		name            string
+		b               []byte
+		initialPayload  []byte
+		wantPayload     []byte
+		wantIsFinalized bool
+		wantStreamID    uint32
+	}{
+		{
```

Comment:
If I'm reading correctly, it doesn't look like these tests exercise large payloads where the payload length indicator is 126 or 127 - I know it might be a bit unwieldy/inefficient to test those scenarios, but it's worth exercising those code paths IMO.

It would also be nice to have a separate test that throws invalid data and checks that it errors gracefully without panics.

Reply:
> invalid data

fuzz tests would be nice for this kind of thing, but yeah for now I can just add some static invalid input